l293x/l293x.go

```diff
@@ -61,44 +61,67 @@ func (d *Device) Stop() {
 // en is the PWM pin that controls the motor speed.
 type PWMDevice struct {
 	a1, a2 machine.Pin
-	en     machine.PWM
+	en     machine.Pin
+	pwm    *machine.TCC
+	ch     uint8
 }
 
 // NewWithSpeed returns a new PWMMotor driver that uses a PWM pin to control speed.
-func NewWithSpeed(direction1, direction2 machine.Pin, speedPin machine.PWM) PWMDevice {
+func NewWithSpeed(direction1, direction2 machine.Pin, speedPin machine.Pin, pwm *machine.TCC) PWMDevice {
 	return PWMDevice{
-		a1: direction1,
-		a2: direction2,
-		en: speedPin,
+		a1:  direction1,
+		a2:  direction2,
+		en:  speedPin,
+		pwm: pwm,
 	}
 }
 
 // Configure configures the PWMDevice.
-func (d *PWMDevice) Configure() {
+func (d *PWMDevice) Configure() error {
 	d.a1.Configure(machine.PinConfig{Mode: machine.PinOutput})
 	d.a2.Configure(machine.PinConfig{Mode: machine.PinOutput})
-	d.en.Configure()
+	err := d.pwm.Configure(machine.PWMConfig{
+		Period: 16384e3, // 16.384ms
```

Comment:
Does driving motors depend on this particular frequency?
If not, a more portable solution would be to let users of this API provide an already configured PWM interface object and channel number so that it is easy to drive multiple motors from a single PWM if the PWM has multiple channels.

Reply:
Done.